language/move-stdlib/sources/type_name.move

```diff
@@ -0,0 +1,29 @@
+/// Functionality for converting Move types into values. Use with care!
+module std::type_name {
+    use std::ascii::String;
+
+    struct TypeName has copy, drop, store {
+        /// String representation of the type. All types are represented
+        /// using their source syntax:
+        /// "u8", "u64", "u128", "bool", "address", "vector", "signer" for ground types.
+        /// Struct types are represented as fully qualified, `0x`-prefixed short type names
+        /// `0x1::string::String` or
+        /// `0xabcd::module_name1::type_name1<0x1234::module_name2::type_name2<u64>>`
+        /// Note: addresses in short type names are *always* shortened by stripping leading 0's
+        /// e.g., 0xA rather than 0x00000000A,
+        name: String
+    }
+
+    /// Return a value representation of the type `T`.
+    public native fun get<T>(): TypeName;
```

Comment:
What is the thought behind the name `get`?
My worry is making it clear that it is allocating something, not possibly fetching from a cache. `new` might be better here from that line of thinking.

Reply:
The reasoning is that you are "extracting" or "getting" the `TypeName` from `T` rather than constructing it (e.g., from a string). Trying to evoke that these come from real types.

Reply:
Makes sense, I think should think about this more then as `T::type_name()` sorta, rather than `TypeName::new()`